### PR TITLE
Use one NavHost (with a single navigation scaffold)

### LINKED ISSUE #96: App uses nested NavHosts with unintuitive routing
This app has two `NavHost`s, one in [`Main.kt`][1] and another (nested) one in [`Home.kt`][2].

One peculiarity is the `chats` route (for the chat list) is in the nested `NavHost`, while the `chat/{chatId}` route is in the outer `NavHost`. It feels opposite from the expected, which is that `chats` should be a parent route for `chat/{chatId}`. Alternatively, to support large screens, there should be a destination capable of showing both `chats` and a selected `chat/{chatId}` destinations side by side.

Also, the nested `NavHost` isn't really used for navigation; there are no `navigate()` calls on its `NavHostController`. Instead, the current destination is external state that is read by `NavHost`, specifically to set [`startDestination`][3] to the route of the currently selected item in the nav bar. This has some undesirable effects:
- The `NavHost` recomposes entirely whenever the selected item changes.
- The `NavHost` is always on the start destination, so pressing back always exits completely instead of taking the user back to Chats first.

Ideally, the app should have one stable nav graph with all its destinations defined.

[1]: https://github.com/android/socialite/blob/85d11e853322b7afc0ca77ff2ac219389289d64f/app/src/main/java/com/google/android/samples/socialite/ui/Main.kt#L82
[2]: https://github.com/android/socialite/blob/85d11e853322b7afc0ca77ff2ac219389289d64f/app/src/main/java/com/google/android/samples/socialite/ui/home/Home.kt#L93
[3]: https://github.com/android/socialite/blob/85d11e853322b7afc0ca77ff2ac219389289d64f/app/src/main/java/com/google/android/samples/socialite/ui/home/Home.kt#L95

### PR DESCRIPTION
**Pull request by jdkoren**
Consolidates all the navigation into one NavHost. A single NavigationSuiteScaffold is used to show top-level routes, with logic to hide some or all forms of navigation based on the current destination.

Fixes #96.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -30,11 +30,12 @@ hilt = "2.51.1"
 hiltNavigationCompose = "1.2.0"
 junit = "4.13.2"
 kotlin = "2.0.0"
+kotlinxSerializationJson = "1.6.3"
 ksp = "2.0.0-1.0.21"
 lifecycle = "2.8.4"
 material3 = "1.2.1"
 media3 = "1.4.1"
-navigation = "2.7.7"
+navigation = "2.8.0-rc01"
```
Comment (alexvanyo) [inline gradle/libs.versions.toml:38]: Nit: since `2.8.1` is released, can we use that?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -30,11 +30,12 @@ hilt = "2.51.1"
 hiltNavigationCompose = "1.2.0"
 junit = "4.13.2"
 kotlin = "2.0.0"
+kotlinxSerializationJson = "1.6.3"
```
Comment (dturner) [inline gradle/libs.versions.toml:33]: Nit: 1.7.3 is the latest

File: gradle/libs.versions.toml (hunk 1 above)
Comment (jdkoren) [reply gradle/libs.versions.toml]: Done

File: gradle/libs.versions.toml (hunk 2 above)
Comment (jdkoren) [reply gradle/libs.versions.toml]: Done

**Review by dturner — APPROVED** [review]

**Review by alexvanyo — APPROVED** [review]

**Pull request merged**